open-metadata-implementation/adapters/open-connectors/rest-client-connectors/spring-rest-client-connector/src/main/java/org/odpi/openmetadata/adapters/connectors/restclients/spring/SpringRESTClientConnector.java

```diff
@@ -41,49 +49,317 @@
+                    if (StringUtils.isNotBlank(keystore)) {
+                        log.debug("egeria-ssl: keystore set from spring server environment " + SYSPROP_SSL_SERVER_KEYSTORE);
+                        keystorePasswordSetting = System.getProperty(SYSPROP_SSL_SERVER_KEYSTOREPASS);
+                    }
+                }
+            }
+        }
+
+        if (keystoreSetting != null)
+        {
+            keystore=keystoreSetting;
+            keystorePassword=keystorePasswordSetting;
+            log.debug("egeria-ssl: keystore SET to " + truststore);
+        }
+        else
+            log.debug("egeria-ssl: keystore left at default");
+
+        // The truststore and password should now be available - null will just mean not certs loaded
+
+        // As of Mar 2020 TLS v1.1 is deprecated. Java 8 supports v1.2, so is our default
+        // Java 11 upwards supports TLS v1.3 so we will enable that if we can
+        if (Float.valueOf(System.getProperty("java.specification.version"))>=11) {
+            protocol = protocol13;
+        }
+
+        log.debug("egeria-ssl: protocol is " + protocol);
+    }
+
     /**
      * Default constructor
      */
-    public SpringRESTClientConnector() throws NoSuchAlgorithmException, KeyStoreException, KeyManagementException
-    {
+    public SpringRESTClientConnector() throws RESTConfigurationException {
+
         super();
 
-        /*
-         * Rather than creating a RestTemplate directly, the RestTemplateBuilder is used so that the
-         * uriTemplateHandler can be specified. The URI encoding is set to VALUES_ONLY so that the
-         * '+' character, which is used in queryParameters conveying searchCriteria, which can be a
-         * regex, is encoded as '+' and not converted to a space character.
-         * Prior to this change a regex containing a '+' character would be split into two space
-         * separated words. For example, the regex "name_0+7" (which would match name_07, name_007,
-         * name_0007, etc) would be sent to the server as "name_0 7".
-         */
-        DefaultUriBuilderFactory builderFactory = new DefaultUriBuilderFactory();
-        builderFactory.setEncodingMode(DefaultUriBuilderFactory.EncodingMode.VALUES_ONLY);
+        final String methodName = "SpringRESTClientConnector";
+        /* TODO: Disable SSL cert verification -- for now */
 
+        // Numerous opportunities for error. We'll propogate the base exception to caller, and it will appear in audit log
+        try {
+            SSLContext sc = SSLContext.getInstance(protocol);
 
-        /* TODO: Disable SSL cert verification -- for now */
-        HttpsURLConnection.setDefaultHostnameVerifier(bypassVerifier);
-        SSLContext sc = SSLContext.getInstance("SSL");
-        sc.init(null, INSECURE_MANAGER, null);
-        HttpsURLConnection.setDefaultSSLSocketFactory(sc.getSocketFactory());
+            TrustManager[] trustManagers = null;
+            KeyManager[] keyManagers = null;
```

Comment:
TrustManager and KeyManager lists per client? I think is not a good idea

Reply:
Not so much the lists, but instances per client I'd agree -  this is indeed inefficient. 

I felt it important to avoid changing the default context to avoid interaction with rest requests for other client libraries. By ensuring we use our own context we are isolated. This is especially important for 3rd party applications that are making use of our libraries. At some point we might expect the client API and/or connector properties to support specification of the configuration, for now they are at least scoped to egeria